oneflow/user/ops/fused_attention_ops.cpp

```diff
@@ -632,4 +632,106 @@ Maybe<void> ParseSplitAxis(const std::string& layout, bool can_hk_split, int64_t
   return Maybe<void>::Ok();
 }
 
+/* static */ Maybe<void> FusedApplyRotaryEmbOp::InferLogicalTensorDesc(user_op::InferContext* ctx) {
+  const user_op::TensorDesc& x_desc = ctx->InputTensorDesc("x", 0);
+  const std::string& x_layout = ctx->Attr<std::string>("x_layout");
+  const std::string& mode = ctx->Attr<std::string>("mode");
+  const int64_t rotary_size = ctx->Attr<int64_t>("rotary_size");
+  const int64_t k_size = ctx->Attr<int64_t>("k_size");
+  const int64_t tensor_index = ctx->Attr<int64_t>("tensor_index");
+
+  CHECK_OR_RETURN((tensor_index >= 0) && (tensor_index <= 2));
+  CHECK_OR_RETURN((mode == "interval") || (mode == "plane"));
+
+  if (ctx->has_input("output_layout", 0)) {
+    const std::string& output_layout = ctx->Attr<std::string>("output_layout");
+
+    if (output_layout != x_layout) {
+      CHECK_OR_RETURN(
+          (output_layout == "BMHK"
+           && (x_layout == "BM(HK)" || x_layout == "BM(H2K)" || x_layout == "BM(H3K)"))
+          || (output_layout == "MBHK"
+              && (x_layout == "MB(HK)" || x_layout == "MB(H2K)" || x_layout == "MB(H3K)")));
+    }
+  }
```

Comment:
这里有问题，output_layout是attr，不是input tensor,  这里用has_input()会一直判断为false, (output_layout需要设置一下默认参数值)
```suggestion
const std::string& output_layout = ctx->Attr<std::string>("output_layout");
if (output_layout && output_layout != x_layout) {
      CHECK_OR_RETURN(
          (output_layout == "BMHK"
           && (x_layout == "BM(HK)" || x_layout == "BM(H2K)" || x_layout == "BM(H3K)"))
          || (output_layout == "MBHK"
              && (x_layout == "MB(HK)" || x_layout == "MB(H2K)" || x_layout == "MB(H3K)")));
  }
```